services/drone/drone-build.service.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+
+const Joi = require('joi')
+const serverSecrets = require('../../lib/server-secrets')
+const { isBuildStatus, renderBuildStatusBadge } = require('../build-status')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  status: Joi.alternatives()
+    .try(isBuildStatus, Joi.equal('none'))
+    .required(),
+}).required()
+
+module.exports = class DroneBuild extends BaseJsonService {
+  static get category() {
+    return 'build'
+  }
+
+  static get route() {
+    return {
+      base: 'drone',
+      pattern: ':scheme(http|https)?/:host/build/:user/:repo/:branch*',
```

Comment:
I assume this pattern is constructed this way to attempt to support Cloud & self-hosted instances conditionally, but I'm not sure that this is the best approach.

With scheme being optional, this is a "valid" route:
https://shields-staging-pr-3240.herokuapp.com/drone/mydrone-deployment.com/build/drone/drone.svg 

that ends up trying to make a protocol-less request to `://mydrone-deployment.com/apis/repos/drone/drone/builds/latest` which obviously is not a valid url. Accordingly it returns an `inaccessible` badge:

![](https://shields-staging-pr-3240.herokuapp.com/drone/mydrone-deployment.com/build/drone/drone.svg)

It would be fine to make the protocol and host mandatory in all cases (users can always enter `https` and `cloud.drone.io` respectively), this pattern is done for other badges (see [Jira](https://github.com/badges/shields/blob/master/services/jira/jira-issue.service.js) for example) . Or if you want to provide a separate route for badges targeting the cloud/SaaS instance to _not_ require users specify those values, then you could split this into two separate service classes (feel free to extract common functionality into separate functions and/or a base class) each with a more explicit route.

Also, let's change the route base to `drone/build` to be consistent with the Shields route conventions

Reply:
And yet another option to consider.. 😄 some of our badge routes use a default endpoint (typically the cloud/SaaS service), and accept a queryParam to support custom self-hosted endpoints. 

[Bitbucket Pull Requests](https://github.com/badges/shields/blob/master/services/bitbucket/bitbucket-pull-request.service.js#L92) is a good example of this. 

With that approach you could change the route here to:
```
  static get route() {
    return {
      base: 'drone/build',
      pattern: ':user/:repo/:branch*',
    }
  }
```

and the routes for build badges for cloud Drone would be:
`https://img.shields.io/drone/build/drone/drone.svg`

and for a self-hosted Drone instance:
`https://img.shields.io/drone/my-user/my-repo/drone.svg?server=https://my-self-hosted-drone.io`

Reply:
Ooh! I think I like the last approach the most! If it's ok with you I'll try to rip off bitbucket tonight 😄 This solution has the best readability vs functionality vs DRY in my opinion.

Reply:
Done! 🚀 Thanks for the detailed options here, it helps a lot understanding the idea/logic within Shields ❤️ Hope you like it!